src/metrics.rs

```diff
@@ -0,0 +1,250 @@
+//! The `metrics` module enables sending measurements to an InfluxDB instance
+
+use std::sync::mpsc::{channel, Receiver, RecvTimeoutError, Sender};
+use std::sync::{Arc, Barrier, Mutex};
+use std::sync::{Once, ONCE_INIT};
+use std::thread;
+use std::time::{Duration, Instant};
+
+pub type Point = String; // TODO: replace with InfluxDB point struct
+
+#[derive(Debug)]
+enum MetricsCommand {
+    Submit(Point),
+    Flush(Arc<Barrier>),
+}
+
+struct MetricsAgent {
+    sender: Sender<MetricsCommand>,
+}
+
+trait MetricsWriter {
+    // Write the points and empty the vector.  Called on the internal
+    // MetricsAgent worker thread.
+    fn write(&self, points: &mut Vec<Point>);
+}
+
+struct InfluxDbMetricsWriter {}
+
+impl InfluxDbMetricsWriter {
+    fn new() -> InfluxDbMetricsWriter {
+        InfluxDbMetricsWriter {}
+    }
+}
+
+impl MetricsWriter for InfluxDbMetricsWriter {
+    fn write(&self, points: &mut Vec<Point>) {
+        // TODO
+        points.clear();
+    }
+}
+
+impl MetricsAgent {
+    fn new() -> MetricsAgent {
+        Self::new_with_args(Arc::new(InfluxDbMetricsWriter::new()), None)
+    }
+
+    fn new_with_args(
+        writer: Arc<MetricsWriter + Send + Sync>,
+        write_frequency: Option<Duration>,
```

Comment:
I'd drop the `Option` here and instead add that 10s default in the constructor above.

Reply:
✔️